File: app/src/main/java/com/itba/runningMate/rundetails/RunDetailsView.java
package com.itba.runningMate.rundetails;

import android.graphics.Bitmap;
import android.net.Uri;

import com.itba.runningMate.domain.Route;
import com.itba.runningMate.rundetails.model.RunMetricsDetail;

public interface RunDetailsView {

    void showRoute(Route route);

    void showRunMetrics(RunMetricsDetail runMetrics);

    void endActivity();

    void shareImageIntent(Uri uri);

    Bitmap getMetricsImage(RunMetricsDetail detail);

    void showShareRunError();

    void showUpdateTitleError();

    void showDeleteError();

    void showRunNotAvailableError();

}

File: app/src/main/java/com/itba/runningMate/rundetails/RunDetailsView.kt
package com.itba.runningMate.rundetails

import android.graphics.Bitmap
import android.net.Uri
import com.itba.runningMate.domain.Route
import com.itba.runningMate.rundetails.model.RunMetricsDetail

interface RunDetailsView {

    fun showRoute(route: Route)

    fun showRunMetrics(runMetrics: RunMetricsDetail)

    fun endActivity()

    fun shareImageIntent(uri: Uri)

    fun getMetricsImage(detail: RunMetricsDetail?): Bitmap?

    fun showShareRunError()

    fun showUpdateTitleError()

    fun showDeleteError()

    fun showRunNotAvailableError()

}